securedrop_client/gui/conversation/export/print_transcript_dialog.py

```diff
@@ -0,0 +1,39 @@
+from PyQt5.QtCore import QSize, pyqtSlot
+
+from securedrop_client.gui.conversation.export import PrintDialog
+
+from .device import Device
+
+
+class PrintTranscriptDialog(PrintDialog):
+    """Adapts the dialog used to print files to allow printing of a conversation transcript.
+
+    - Ajust the init arguments to the needs of conversation transcript printing.
+    - Adds a method to allow a transcript to be printed.
+    - Overrides the slot that handles the printing action to call said method.
+    """
+
+    def __init__(self, device: Device, file_name: str, transcript_location: str) -> None:
+        super().__init__(device, "", file_name)
+
+        self.transcript_location = transcript_location
+
+    def _print_transcript(self) -> None:
+        self._device.print_transcript(self.transcript_location)
+        self.close()
+
+    @pyqtSlot()
+    def _on_print_preflight_check_succeeded(self) -> None:
+        # If the continue button is disabled then this is the result of a background preflight check
```

Comment:
Sadly, that's not what the comment means. (The indirectness of this behavior makes me consider it the biggest smell in this dialog implementation.)

The following is not true: every time a background preflight check is in progress the button is disabled. Or rather, there is no relation of causality between those two facts.

It happens that the button is disabled during a first period of time. A preflight check is triggered automatically during that time. That's what is referred to in the comment as a _background_ preflight check (one that's not triggered by user behavior).

Once the result of that preflight check is known, the button is enabled and a second period of time starts. During that second period of time, the button is configured to trigger a preflight check. When the response to that preflight check becomes known, the button will happen to be enabled.

What the comment is really meaning in my words is: _if the button is disabled, then the step 1 of the wizard hasn't finished yet, so the next step is step 2. Otherwise, the next step is step 3._ The button is an observable side-effect of steeping through the wizard. The state of the button _can_ (didn't say _should_!) be used to keep track of the progression through the different steps, and changes of step happen as a result of preflight checks. But the fact that the button can be used in that way is really just an observation, not a necessity. (I was going to say it's a coincidence, but not really. If the button state wasn't an way to discriminate between wizard steps, something else _could_ (didn't say _should_!) be used.)

In a more robust implementation, the state of the wizard would be explicit. (Which is what the `QWizard` class allows, via separate _pages_, but this is a `QDialog` over-stretched to implement a wizard.)

Reply:
I let you decide what's best to to here. I would leave the comment as-is, because fixing the cause of the complexity is out-of-scope and the comment has the property of calling the attention to it. (:grimacing:) Also, I don't believe there will be any maintenance required on that specific topic. (If we were to touch the sequence of states of these dialogs again, we should turn them into proper wizards IMO, _cough cough_.)

All that being said, if you can find a better comment, and feel like adding it, please do!

Reply:
Thanks very much for unpacking the current behavior behind this comment, @gonzalo-bulnes.  _Given_ that behavior, this comment makes sense, is sufficient, and indeed is maybe the most matter-of-fact articulation possible (!).